### PR TITLE
feat: (frontend) Sort-Show answers in HomePage.

### LINKED ISSUE #84: Show/Sort answer in the homepage.
Migrate the functionality of this [ticket](https://github.com/wizeline/wize-q-front/issues/760) of the old project 

Old Tickets Related : 
[Front](https://github.com/wizeline/wize-q-front/issues/760)
[Back](https://github.com/wizeline/wize-q-back/issues/520)

=================================================================================================
Show in the Homepage the answer according to the next criteria:              

- Official answer
- Comment approved as answer
- Community Answer (votes)

### 🚨   **Only one of the options should be shown.**

Proposal: 
> We could use this component which is hardcoded right now to display "Official Answer"/"Approved an answer"/"Community Answer"  if its the case in the list questions/home page.

image

You can see a full conversation about this [here](https://github.com/wizeline/wize-q-back/pull/499).

### PR DESCRIPTION
**Pull request by GeorginaFraire**
#### What does this PR do?
(A brief explanation synthesizing the feature, bug or fix.)
-  Show the different type of answer for a question.

> If there is an official answer, show it on the homepage, if not, search for a comment approved by the admin, if it exists, show it on the homepage and if it is the last one not exists, search for a comment marked as a community response, if it exists, show it on the homepage.
> If none of the above options exist, it should not show anything.

#### How should this be manually tested?
1. Login
2. If a question has an answer, verify that you can see in the homepage
3. If a question has an approved comment as an answer and no official answer, the comment should show as an answer labeled "Comment Approved By ......" instead of "Best Answer".
4. If a question has a comment marked as a community answer and does not have an official answer or admin-approved comment, the comment marked as an answer should be displayed on the homepage as an answer and with the legend “Community Answer”.
5. Go to the end of the page and verify that each question load correctly. 
6. 🎉 

#### What are the relevant tickets?
(Link to issues, related PR, JIRA issues, etc.)
Closes #84 

#### Screenshots
Answers

### CODE REVIEW

**Comment by AlonsoMtzG** [comment]
I added some things to the Label component so that when the answer is 'Approved', it has the same style as it already had in the comments section

File: app/components/AnswerRow/AnswerRow.jsx [hunk 1]
```diff
@@ -40,15 +39,29 @@ function AnswerRow({ searchTerm, isPreview, isQuestionModalOpen, ...props }) {
...
+      txt_answer = `Comment approved by ${props.approver.full_name}`;
+    }
+    else if (props.isCommunityAnswer) {
+      txt_answer = 'Community answer';
+    };
+    
+    return (
+      <Label type={'Answer'} text={txt_answer} />
+    )
+  }
```
Comment (AlonsoMtzG) [inline app/components/AnswerRow/AnswerRow.jsx:55]: With the previous mentioned changes can do something like this:
```suggestion
  const renderAnswerLabel = () => {
    if(props.isCommentApproved) {
      return <Label type={'Answer'} text={'Approved'} approvedBy={props.approver.full_name} />
    }
    else if (props.isCommunityAnswer) {
      return <Label type={'Answer'} text={'Community answer'} />
    };
    return (
      <Label type={'Answer'} text={'Best Answer'} />
    )
  }
```

File: app/components/QuestionCard/QuestionCard.jsx [hunk 2]
```diff
@@ -62,6 +66,38 @@ const QuestionCard = (props) => {
...
+      commentAsAnswer = question.Comments.find(comment => comment.approvedBy !== null);
+    }
+    else if(question.hasCommunityAnswer){
+      const [communityAnswerCommentId, ] = reorderHighlightedComments(question.Comments);
+      commentAsAnswer =  question.Comments.find(comment => comment.id === communityAnswerCommentId);
+    }
+
+    if(commentAsAnswer){
+      return (<AnswerRow 
+      answer_text= {commentAsAnswer.comment}
```
Comment (Hoosep) [inline app/components/QuestionCard/QuestionCard.jsx:86]:
```suggestion
      answer_text={commentAsAnswer.comment}
```

File: app/components/QuestionCard/QuestionCard.jsx [hunk 3]
```diff
@@ -62,6 +66,38 @@ const QuestionCard = (props) => {
...
+      user={commentAsAnswer.User}
+      answered_at={commentAsAnswer.createdAt}
+      searchTerm={renderAnswerProps.searchTerm}
+      isPreview={renderAnswerProps.isPreview}
+      isFromList={renderAnswerProps.isFromList}
+      questionId={question.question_id}
+      isAnswer={question.Answers.length > 0}
+      isCommunityAnswer= {question.hasCommunityAnswer}
```
Comment (Hoosep) [inline app/components/QuestionCard/QuestionCard.jsx:94]:
```suggestion
      isCommunityAnswer={question.hasCommunityAnswer}
```

File: app/components/QuestionCard/QuestionCard.jsx [hunk 4]
```diff
@@ -62,6 +66,38 @@ const QuestionCard = (props) => {
...
+      isCommentApproved ={question.hasCommentApproved}
```
Comment (Hoosep) [inline app/components/QuestionCard/QuestionCard.jsx:95]:
```suggestion
      isCommentApproved={question.hasCommentApproved}
```

File: app/components/QuestionCard/QuestionCard.jsx [hunk 5]
```diff
@@ -62,6 +66,38 @@ const QuestionCard = (props) => {
...
+      approver ={commentAsAnswer.Approver}
```
Comment (Hoosep) [inline app/components/QuestionCard/QuestionCard.jsx:96]:
```suggestion
      approver={commentAsAnswer.Approver}
```

**Review by Hoosep — APPROVED** [review]
I left you some suggestions but my review is already approved :)

File: app/components/AnswerRow/AnswerRow.jsx (hunk 1 above)
Comment (AlonsoMtzG) [reply app/components/AnswerRow/AnswerRow.jsx:55]: Would look like this:

Captura de Pantalla 2022-11-16 a la(s) 14 31 50

File: app/components/QuestionCard/QuestionCard.jsx (hunk 5 above)
Comment (GeorginaFraire) [reply app/components/QuestionCard/QuestionCard.jsx:96]: Done!

File: app/components/QuestionCard/QuestionCard.jsx (hunk 4 above)
Comment (GeorginaFraire) [reply app/components/QuestionCard/QuestionCard.jsx:95]: Done!

File: app/components/QuestionCard/QuestionCard.jsx (hunk 2 above)
Comment (GeorginaFraire) [reply app/components/QuestionCard/QuestionCard.jsx:86]: Done!

File: app/components/AnswerRow/AnswerRow.jsx (hunk 1 above)
Comment (GeorginaFraire) [reply app/components/AnswerRow/AnswerRow.jsx:55]: Done!

**Comment by GeorginaFraire** [comment]
Changes suggested implemented. 
Thanks! 

Screen Shot 2022-11-16 at 14 44 21

**Review by AlonsoMtzG — APPROVED** [review]
LGTM 🎉

**Pull request merged**